Added cloudevent signature type using cloudevents sdk-python (#78)
GoogleCloudPlatform/functions-framework-python: README.md

| `--host`           | `HOST`                    | The host on which the Functions Framework listens for requests. Default: `0.0.0.0`                                                                                                               |
| `--port`           | `PORT`                    | The port on which the Functions Framework listens for requests. Default: `8080`                                                                                                                  |
| `--target`         | `FUNCTION_TARGET`         | The name of the exported function to be invoked in response to requests. Default: `function`                                                                                                     |

> | `--signature-type` | `FUNCTION_SIGNATURE_TYPE` | The signature used when writing your function. Controls unmarshalling rules and determines which arguments are used to invoke your function. Default: `http`; accepted values: `http` or `event` or `cloudevent` |

Reviewer: ```suggestion
| `--signature-type` | `FUNCTION_SIGNATURE_TYPE` | The signature used when writing your function. Controls unmarshalling rules and determines which arguments are used to invoke your function. Default: `http`; accepted values: `http`, `event` or `cloudevent` |
```

| `--source`         | `FUNCTION_SOURCE`         | The path to the file containing your function. Default: `main.py` (in the current working directory)                                                                                             |
| `--debug`          | `DEBUG`                   | A flag that allows to run functions-framework to run in debug mode, including live reloading. Default: `False`                                                                                   |

# Enable Google Cloud Functions Events
The Functions Framework can unmarshall incoming
Google Cloud Functions [event](https://cloud.google.com/functions/docs/concepts/events-triggers#events) payloads to `data` and `context` objects.
These will be passed as arguments to your function when it receives a request.
Note that your function must use the `event`-style function signature:

```python
def hello(data, context):
    print(data)
    print(context)
```

To enable automatic unmarshalling, set the function signature type to `event`

> using a command-line flag or an environment variable. By default, the HTTP

Reviewer: ```suggestion
 using the `--signature-type` command-line flag or the `FUNCTION_SIGNATURE_TYPE` environment variable. By default, the HTTP
```
Not sure why we lost this. It's included in the `cloudevent` section below.

signature will be used and automatic event unmarshalling will be disabled.

> For more details on this signature type, check out the Google Cloud Functions

Reviewer: ```suggestion
For more details on this signature type, see the Google Cloud Functions
```

documentation on
[background functions](https://cloud.google.com/functions/docs/writing/background#cloud_pubsub_example).

# Enable CloudEvents

> The Functions framework can also unmarshall incoming [CloudEvents](http://cloudevents.io) payloads to the `cloudevent` object. This will be passed as a [cloudevent](https://github.com/cloudevents/sdk-python) to your function when it receives a request.  Note that your function must use the cloudevents-style function signature:

Reviewer: ```suggestion
The Functions framework can also unmarshall incoming [CloudEvents](http://cloudevents.io) payloads to the `cloudevent` object. This will be passed as a [cloudevent](https://github.com/cloudevents/sdk-python) to your function when it receives a request.  Note that your function must use the `cloudevents`-style function signature:
```

| `--target`         | `FUNCTION_TARGET`         | The name of the exported function to be invoked in response to requests. Default: `function`                                                                                                     |
| `--signature-type` | `FUNCTION_SIGNATURE_TYPE` | The signature used when writing your function. Controls unmarshalling rules and determines which arguments are used to invoke your function. Default: `http`; accepted values: `http` or `event` or `cloudevent` |
| `--source`         | `FUNCTION_SOURCE`         | The path to the file containing your function. Default: `main.py` (in the current working directory)                                                                                             |
| `--debug`          | `DEBUG`                   | A flag that allows to run functions-framework to run in debug mode, including live reloading. Default: `False`                                                                                   |

> 

Reviewer 2: Signature-type of `event` is not a CloudEvent. The documentation is wrong here.

Can we update this section to not say CloudEvents?

Here is the section in Node for your reference:
https://github.com/GoogleCloudPlatform/functions-framework-nodejs#enable-google-cloud-functions-events

Author: Resolved!

    print(context)
```

> 

Reviewer 2: (Similar comment above) The above `data, context` is not a CloudEvent, it's a GCF event. We should change this section as such.

The Functions framework can also unmarshall incoming [CloudEvents](http://cloudevents.io) payloads to the `cloudevent` object. This will be passed as a [cloudevent](https://github.com/cloudevents/sdk-python) to your function when it receives a request.  Note that your function must use the cloudevents-style function signature:

```python
def hello(cloudevent):
    print(f"Received event with ID: {cloudevent['id']}")
```

> To enable automatic unmarshalling, set the function signature type to either `event` or `cloudevent` using the `--signature-type` command-line flag or the `FUNCTION_SIGNATURE_TYPE` environment variable. By default, the HTTP signature type will be used and automatic event unmarshalling will be disabled.

Reviewer 2: Only the `cloudevent` signature should support CloudEvents. The `events` signature should not support CloudEvents.

Author: Ah I see. Removed!
